groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -765,787 +1431,10708 @@ void Optional_DataImp<TYPE>::emplace(
+    template <class... ARGS>
+    void emplace(BSLS_COMPILERFEATURES_FORWARD_REF(ARGS)...);
+
+#if defined(BSLS_COMPILERFEATURES_SUPPORT_GENERALIZED_INITIALIZERS)
+    template <class INIT_LIST_TYPE, class... ARGS>
+    void emplace(std::initializer_list<INIT_LIST_TYPE>,
+                                        BSLS_COMPILERFEATURES_FORWARD_REF(ARGS)
+                                        ...);
+#endif
+// }}} END GENERATED CODE
+#endif
+    void reset() BSLS_KEYWORD_NOEXCEPT;
+        // Reset this object to the default constructed state (i.e., to a
+        // disengaged state).
+
+    void swap(optional& other);
+        // Efficiently exchange the value of this object with the value of the
+        // specified 'other' object.  In effect, performs
+        // 'using bsl::swap; swap(c, other.c);'.
+
+#ifdef BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+    TYPE&  value() &;
+    TYPE&& value() &&;
+        // Return a reference providing modifiable access to the underlying
+        // 'TYPE' object.  Throws a 'bsl::bad_optional_access' if the
+        // 'optional' object is disengaged.
+#else
+    TYPE& value();
+        // Return a reference providing modifiable access to the underlying
+        // 'TYPE' object.  Throws a 'bsl::bad_optional_access' if the
+        // 'optional' object is disengaged.
+
+#endif  // BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+
+#ifdef BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+    template <class ANY_TYPE>
+    TYPE value_or(ANY_TYPE&& rhs) &&;
+        // If 'this->has_value() == true', return a copy of the contained
+        // 'value_type' object.  Otherwise, return the value of specified 'rhs'
+        // converted to 'value_type'.
+
+#ifdef BSL_COMPILERFEATURES_GUARANTEED_COPY_ELISION
+    template <class ANY_TYPE>
+    TYPE value_or(bsl::allocator_arg_t, allocator_type, ANY_TYPE&&) &&;
+      // If 'this->has_value() == true', return a copy of the contained
+      // 'value_type' object created using the provided allocator.  Otherwise,
+      // return the value of specified 'rhs' converted to 'value_type' and use
+      // the specified 'allocator' for the returned object.
+#endif  //BSL_COMPILERFEATURES_GUARANTEED_COPY_ELISION
+#endif  //BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+
+    optional& operator=(bsl::nullopt_t) BSLS_KEYWORD_NOEXCEPT;
+        // Reset this object to be disengaged and return a reference providing
+        // modifiable access to this object.
+
+    optional& operator=(const optional& rhs);
+    optional& operator=(BloombergLP::bslmf::MovableRef<optional> rhs);
+        // If specified 'rhs' is engaged, assign its value to this object.
+        // Otherwise, reset this object to a disengaged state.  Return a
+        // reference providing modifiable access to this object.  Note that
+        // this method may invoke assignment from 'rhs', or construction from
+        // 'rhs', depending on whether this 'optional' object is engaged.
```

Comment:
Our special member functions typically have a fairly standardized doc.  Please copy with appropriate modifications (if any) from `baltzo_localtimedescriptor`.

Reply:
assignment operators modified to resemble baltzo_localtimedescriptor and, where appropriate, nullablevalue